### PR TITLE
Helper class to allow in process servers to use peer uids in test

### PR DESCRIPTION
**Pull request by abtom**
We  use in process servers for testing which currently does not allow the use peer uids due to InProcessBuilder not setting the correct values. Adding a test only interceptor so that we can use peer uids in  in process servers.

### CODE REVIEW

**Comment by abtom** [comment]
See cl/613744041

**Comment by jdcormie** [comment]
Can you update this to the latest piper revision ?

Also please move the PeerUidTestHelper from src/main/ to src/testFixtures/. PeerUidTestHelperTest should stay in src/test/

**Comment by abtom** [comment]
updated

**Comment by jdcormie** [comment]
Can you make sure the whitespace and formatting complies with 
https://github.com/grpc/grpc-java/blob/master/CONTRIBUTING.md#code-style ?

File: binder/build.gradle [hunk 1]
```diff
@@ -56,6 +56,7 @@ dependencies {
     testImplementation project(':grpc-testing')
     testImplementation project(':grpc-inprocess')
     testImplementation testFixtures(project(':grpc-core'))
+    testImplementation project(':grpc-binder')
```
Comment (jdcormie) [inline binder/build.gradle:59]: I'm surprised that you need this dep ... ?

Comment (abtom) [reply binder/build.gradle:59]: I don't think `PeerUidTestHelperTest` can see `PeerUidTestHelper` without it.

It was failing to build

https://github.com/grpc/grpc-java/actions/runs/8352583841/job/22862916530

Comment (jdcormie) [reply binder/build.gradle:59]: Hmm I couldn't see a build error. I do see:
```
> Task :grpc-servlet:undertowTest

io.grpc.servlet.UndertowTransportTest > basicStream FAILED
    java.lang.AssertionError at Assert.java:89
```

but I think that is just a flaky test.

The reason I ask is that we already have two tests that successfully depend on testFixtures/...

Comment (abtom) [reply binder/build.gradle]: Ah noted. Reverted the change.

**Review by jdcormie — APPROVED** [review]

**Pull request merged**